db/database.go

```diff
@@ -162,6 +162,13 @@ func (d *PgxDatabase) Transact(fn TxFunc, opts ...TxOpt) (interface{}, error) {
 	}
 	var attempt int
 	var ret interface{}
+	var tx pgx.Tx
+	defer func() {
```

Comment:
Can you just move this to after we open the transaction, below?  That way we can just use tx, and I think the code becomes more clear.

Reply:
using defer in a loop can be a problem - we would like to have this one once or scope the operation into a different function and make sure it is committed there